lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_temporal_extent.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'nokogiri'
+require 'adiwg/mdtranslator/internal/internal_metadata_obj'
+require_relative 'module_time_period'
+
+module ADIWG
+   module Mdtranslator
+      module Readers
+         module Iso191152
+            module TemporalExtent
+               @@temporalExtXPath = 'gmd:EX_TemporalExtent'
+               @@timePeriodXPath = './/gml:TimePeriod'
+               def self.unpack(xtemporalelem, hResponseObj)
+                  intMetadataClass = InternalMetadata.new
+                  hTemporalExt = intMetadataClass.newTemporalExtent
+
+                  xTemporalExt = xtemporalelem.xpath(@@temporalExtXPath)[0]
+                  return nil if xTemporalExt.nil?
```

Comment:
I'm still unclear the times we return `nil` vs, in this case, `hTemporalExt`

Reply:
i've been returning `nil` to basically represent a no-go for `self.unpack`. I think it's simpler to understand compared to an empty hash. I know johnathan mentioned making sure to return the hash in situations to make the writers happy. I was planning on getting an exact list of which modules have to do that vs those that don't in this next iteration.